File: frontend/edenBooks/src/modules/users/services/userService.ts

import { userApi } from "../api/userApi";
import { IUserRequest } from "../interfaces/IUserRequest";
import { IUserService } from "../interfaces/IUserService";

export const userService:IUserService= {
  Login: async (email: string, password: string) => userApi.Login(email, password),
  Register: async (email: string, password: string) => userApi.Register(email, password),
  GetUserById: async (id: number) => userApi.GetUserById(id),
  UpdateUser: async (id: number, user: IUserRequest) =>userApi.UpdateUser(id, user),
  
  
};

File: backend/internal/modules/users/domain/ports/services/user_service.go
package services

import (
	"github.com/javierjpv/edenBooks/internal/modules/users/domain/entities"
)

type UserService interface {
	Register(user entities.User) error
	Login(email string, password string) (*entities.User, error)
	GetUserByID(id uint) (*entities.User, error)
	UpdateUser(u *entities.User) error
}
